backend/services/grid/grid_service.py

```diff
@@ -57,6 +62,34 @@ def trim_grid_to_aoi(grid_dto: GridDTO) -> geojson.FeatureCollection:
                 intersecting_features.append(clipped_feature)
         return geojson.FeatureCollection(intersecting_features)
 
+    @staticmethod
+    def trim_aoi_to_roads(grid_dto: GridDTO) -> geojson.FeatureCollection:
+        """
+        Extension of trim_grid_to_aoi. Further trims the grid to only those with roads
+        :param grid_dto: the dto containing
+        :return: geojson.FeatureCollection trimmed task grid
+        """
+        trimmed_grid = GridService.trim_grid_to_aoi(grid_dto)
+        overarching_bbox = GridService._create_overarching_bbox(grid_dto)
+        # overarching_bbox = (32.568389314727,-1.9150114059448,32.574871994676,-1.9065248966217)
+        roads = []
+
+        url = 'https://overpass-api.de/api/interpreter?data=[out:json][timeout:25];(node["highway"]{};way["highway"]{};relation["highway"]{};);out geom;>;out skel qt;'.format(
+            overarching_bbox, overarching_bbox, overarching_bbox
+        )
```

Comment:
This does the same thing just a little easier to parse for the reader

```suggestion
        url = 'https://overpass-api.de/api/interpreter?data=[out:json][timeout:25];(node["highway"]{bbox};way["highway"]{bbox};relation["highway"]{bbox};);out geom;>;out skel qt;'.format(
                bbox=overarching_bbox,
        )
```

Reply:
Didn't even know you could do that! Today I learned.